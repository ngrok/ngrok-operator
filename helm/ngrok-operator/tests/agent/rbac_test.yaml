suite: test agent rbac
templates:
- agent/rbac.yaml
tests:
- it: Should match snapshot
  asserts:
  - matchSnapshot: {}
- it: should create a clusterrole
  documentIndex: 0
  asserts:
  - isKind:
      of: ClusterRole
  - isAPIVersion:
      of: rbac.authorization.k8s.io/v1
- it: should create a clusterrolebinding
  documentIndex: 1
  asserts:
  - isKind:
      of: ClusterRoleBinding
  - isAPIVersion:
      of: rbac.authorization.k8s.io/v1
