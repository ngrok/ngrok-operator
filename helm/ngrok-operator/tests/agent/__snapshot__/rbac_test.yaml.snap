Should match snapshot:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: RELEASE-NAME-ngrok-operator-agent-role
    rules:
      - apiGroups:
          - ingress.k8s.ngrok.com
        resources:
          - tunnels
        verbs:
          - get
          - list
          - watch
          - patch
          - update
      - apiGroups:
          - ingress.k8s.ngrok.com
        resources:
          - tunnels/finalizers
        verbs:
          - update
      - apiGroups:
          - ingress.k8s.ngrok.com
        resources:
          - tunnels/status
        verbs:
          - get
          - patch
          - update
      - apiGroups:
          - ""
        resources:
          - events
        verbs:
          - create
          - patch
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: RELEASE-NAME-ngrok-operator-agent-rolebinding
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: RELEASE-NAME-ngrok-operator-agent-role
    subjects:
      - kind: ServiceAccount
        name: RELEASE-NAME-ngrok-operator-agent
        namespace: NAMESPACE
